metrics-core/src/main/java/software/amazon/swage/metrics/record/cloudwatch/CloudWatchRecorder.java

```diff
@@ -195,13 +261,54 @@ public CloudWatchRecorder(
      *                         Suggested value of 60, for one minute aggregation.
      * @param dimensionMapper Configuration specifying which dimensions to sent
      *                        to CloudWatch for each metric event.
+     * @deprecated Use {@link CloudWatchRecorder.Builder} instead
      */
+    @Deprecated
     public CloudWatchRecorder(
             final AmazonCloudWatch client,
             final String namespace,
             final int maxJitter,
             final int publishFrequency,
-            final DimensionMapper dimensionMapper)
+            final DimensionMapper dimensionMapper) {
+        this(
+                client,
+                namespace,
+                maxJitter,
+                publishFrequency,
+                dimensionMapper,
+                Executors.newSingleThreadScheduledExecutor()
+        );
+    }
+
+    /**
+     * Create a new recorder instance.
+     * This recorder will periodically send aggregated metric events to CloudWatch
+     * via the provided client. Requests will be queued and sent using a
+     * single-threaded ScheduledExecutorService every publishFrequency (in seconds).
+     * The initial submission to CloudWatch will be delayed by a random amount
+     * on top of the publish frequency, bounded by maxJitter.
+     *
+     * The {@link #shutdown} method must be called when the application is done
+     * with the recorder in order to flush and stop the reporting thread.
+     *
+     * @param client Client to use for connecting to CloudWatch
+     * @param namespace CloudWatch namespace to publish under
+     * @param maxJitter Maximum delay before counting publish frequency for
+     *                  initial request, in seconds.
+     *                  A value of 0 will provide no jitter.
+     * @param publishFrequency Batch up and publish at this interval, in seconds.
+     *                         Suggested value of 60, for one minute aggregation.
+     * @param dimensionMapper Configuration specifying which dimensions to sent
+     *                        to CloudWatch for each metric event.
+     * @param scheduledExecutorService Executor to schedule metric publishing at a fixed rate.
+     */
+    private CloudWatchRecorder(
```

Comment:
In the same vein does this need to be private? Potential cost of it being public is?

Builder does no null or error checking, so I don't think exposing this increases the risk that customers will use this in error.

Reply:
Similar to my other comment, if we're going to support both constructors and the builder moving forward, this can be made public.